india_compliance/gst_india/report/gstr_3b_details/gstr_3b_details.py

```diff
@@ -474,6 +478,62 @@ def get_for_purchase_invoice(self, group_by="name"):
 
         return self.get_ineligible_credit(credit_availed, credit_available, group_by)
 
+    def get_ineligible_itc_due_to_pos_for_purchase(self, group_by="name"):
+        """
+        - ITC restricted due to PoS rules
+        """
+        ineligible_transactions = self.get_vouchers_with_gst_expense("Purchase Invoice")
+
+        if not ineligible_transactions:
+            return []
+
+        pi = frappe.qb.DocType("Purchase Invoice")
+        taxes = frappe.qb.DocType("Purchase Taxes and Charges")
+
+        # utility function
+        def get_tax_case_statement(account, alias):
+            return Sum(
+                Case()
+                .when(
+                    taxes.account_head.isin(account),
+                    taxes.base_tax_amount_after_discount_amount,
+                )
+                .else_(0)
+            ).as_(alias)
+
+        # Credit availed is not required as it will be always 0 for pos
+
+        ineligible_credit = (
+            frappe.qb.from_(pi)
+            .inner_join(taxes)
+            .on(pi.name == taxes.parent)
+            .select(
+                pi.name.as_("voucher_no"),
+                pi.posting_date,
+                pi.ineligibility_reason.as_("itc_classification"),
+                get_tax_case_statement([self.gst_accounts.igst_account], "iamt"),
+                get_tax_case_statement([self.gst_accounts.cgst_account], "camt"),
+                get_tax_case_statement([self.gst_accounts.sgst_account], "camt"),
+                get_tax_case_statement(
+                    [
+                        self.gst_accounts.cess_account,
+                        self.gst_accounts.cess_non_advol_account,
+                    ],
+                    "csamt",
+                ),
+            )
+            .where(taxes.account_head.isin(list(self.gst_accounts.values())))
+            .where(
+                IfNull(pi.ineligibility_reason, "") == "ITC restricted due to PoS rules"
+            )
+            .where(pi.name.isin(ineligible_transactions))
+            .where(pi.company_gstin != IfNull(pi.supplier_gstin, ""))
```

Comment:
These checks are not required in ineligibility functions. Were recently added, but got out of my sight.